pyk/src/tests/unit/kast/test_inner.py

```diff
@@ -40,3 +40,28 @@ def test_flatten_label(label: str, kast: KInner, expected: list[KInner]) -> None
 
     # Then
     assert actual == expected
+
+
+KEEP_VARS_SORTED_DATA: Final[tuple[tuple[dict[str, list[KVariable]], dict[str, KVariable]], ...]] = (
+    (
+        {'a': [KVariable('a'), KVariable('a')], 'b': [KVariable('b'), KVariable('b')]},
+        {'a': KVariable('a'), 'b': KVariable('b')},
+    ),
+    (
+        {'a': [KVariable('a', 'K'), KVariable('a', 'X')], 'b': [KVariable('b', 'K'), KVariable('b', 'X')]},
+        {'a': KVariable('a'), 'b': KVariable('b')},
+    ),
+    (
+        {'a': [KVariable('a', 'K'), KVariable('a')], 'b': [KVariable('b', 'K'), KVariable('b', 'K')]},
+        {'a': KVariable('a', 'K'), 'b': KVariable('b', 'K')},
+    ),
```

Comment:
```suggestion
    ),
    (
        {'a': [KVariable('a', 'A'), KVariable('a'), KVariable('a', 'B')]},
        {'a': KVariable('a')},
    ),
```

Anotehr test-case to add, that the `break` is working properly.